internal/emscripten/emscripten.go

```diff
@@ -89,8 +110,61 @@ func (v *InvokeFunc) Call(ctx context.Context, mod api.Module, stack []uint64) {
 		panic(err)
 	}
 
-	err = f.CallWithStack(ctx, stack)
+	// The Go implementation below mimics the Emscripten JS behaviour to support
+	// longjmps from indirect function calls. The implementation of these
+	// indirection function calls in Emscripten JS is like this:
+	//
+	// function invoke_iii(index,a1,a2) {
+	//  var sp = stackSave();
+	//  try {
+	//    return getWasmTableEntry(index)(a1,a2);
+	//  } catch(e) {
+	//    stackRestore(sp);
+	//    if (e !== e+0) throw e;
+	//    _setThrew(1, 0);
+	//  }
+	//}
+
+	// This is the equivalent of "var sp = stackSave();".
+	// We reuse savedStack to save allocations.
+	var savedStack [2]uint64
+	err = mod.ExportedFunction("stackSave").CallWithStack(ctx, savedStack[:])
```

Comment:
is `stackSave` exported since the beginning? If this was introduced recently, I am concerned that this might break the existing codes

Reply:
Yes it was, it's not a recent addition to Emscripten, it's just something that we missed during the implementation of our invoke_xxx method since it worked for us in most cases, we just missed the case where a longjmp was used inside the invoke method, which is often used as a throw/catch mechanism in C, all of this is needed to make that work.

Reply:
good to know!